### ISSUE TITLE
Throws WebSocketException when launching Chrome

### ISSUE THREAD

**Issue by lindsve** [kblok/puppeteer-sharp#1029]
### Description

I haven't tried Puppeteer before and I'm currently struggling a bit with the example code because it fails to launch chromium instance on macOS (throws WebSocketException).

### Complete minimal example reproducing the issue

`dotnet new console` and,

```
static void Main(string[] args)
{
    var foo = FoobarAsync().Result;
}

static async Task<string> FoobarAsync()
{
    var browserFetcher = new BrowserFetcher();
    await browserFetcher.DownloadAsync(BrowserFetcher.DefaultRevision);
    var launchOptions = new LaunchOptions
    {
        ExecutablePath = browserFetcher.GetExecutablePath(BrowserFetcher.DefaultRevision),
        Headless = true
    };
    using (var browser = await Puppeteer.LaunchAsync(launchOptions))
    {
        var page = await browser.NewPageAsync();
        await page.GoToAsync("http://localhost:5000");
        var pdfOptions = new PdfOptions { Format = PaperFormat.A4, HeaderTemplate = "<h1>HeaderTemplate</h1>", FooterTemplate = "<h1>FooterTemplate</h1>" };
        var pdfData = await page.PdfStreamAsync(pdfOptions);
        return "";
    }
}
```

### Expected behavior:

No exceptions thrown

### Actual behavior:

I get an exception as soon as it tries to launch Chromium (`... Puppeteer.LaunchAsync(..)`). It seems to download Chromium fine, and there is no change if I point `ExecutablePath` to a local instance of Chrome.

```
Unhandled Exception: System.AggregateException: One or more errors occurred. (Failed to create connection) ---> PuppeteerSharp.ChromiumProcessException: Failed to create connection ---> System.Net.WebSockets.WebSocketException: Unable to connect to the remote server
   at System.Net.WebSockets.WebSocketHandle.ConnectAsyncCore(Uri uri, CancellationToken cancellationToken, ClientWebSocketOptions options)
   at System.Net.WebSockets.ClientWebSocket.ConnectAsyncCore(Uri uri, CancellationToken cancellationToken)
   at PuppeteerSharp.Transport.WebSocketTransport.CreateDefaultWebSocket(Uri url, IConnectionOptions options, CancellationToken cancellationToken)
   at PuppeteerSharp.Transport.WebSocketTransport.CreateDefaultTransport(Uri url, IConnectionOptions connectionOptions, CancellationToken cancellationToken)
   at PuppeteerSharp.Connection.Create(String url, IConnectionOptions connectionOptions, ILoggerFactory loggerFactory, CancellationToken cancellationToken)
   at PuppeteerSharp.Launcher.LaunchAsync(LaunchOptions options)
   --- End of inner exception stack trace ---
   at PuppeteerSharp.Launcher.LaunchAsync(LaunchOptions options)
   at PuppeteerSharp.Launcher.LaunchAsync(LaunchOptions options)
   at pdf_console.Program.FoobarAsync() in /Users/lgeir/src/pdf-console/Program.cs:line 25
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at pdf_console.Program.Main(String[] args) in /Users/me/src/pdf-console/Program.cs:line 13
```

### Versions

* PuppeteerSharp 1.12.1
* .NET Core 2.2.103
* OS: macOS Mojave (10.14.3)

**Comment by kblok** [comment]
Hey @lindsve are you able to run the chromium app downloaded in `.local-chromium`?

**Comment by lindsve** [comment]
Yeah, it launches fine. I do get the macOS dialog asking if I want to allow incoming network requests to Chromium. Accepting that does not change the outcome though

**Comment by kblok** [comment]
MacOS is my main platform for Puppeteer. I never got that.

**Comment by lindsve** [comment]
I browsed through a number of issues and saw that you're using macOS.

I'm not too familiar with the behind-the-scenes requirements here, but could it be isolated to this being a Mac managed in a corporate environment? I could try it again later today on my personal Mac at home.

**Comment by lindsve** [comment]
Seems to narrow it down a bit. It seems to work if I run the app with sudo

`sudo dotnet run`

**Comment by lindsve** [comment]
Finally found the cause for this error. I have set `http_proxy`/`https_proxy`/`no_proxy` environment variables in my user profile. The problem goes away when I unset these.

**Comment by lindsve** [comment]
Solution: Added `127.0.0.1` to `no_proxy` environment variable in addition to `localhost` and other internal domains

**Comment by EdgeHeman** [comment]
I was getting the same error with below line of code on Windows machine
`var browser = await Puppeteer.LaunchAsync(new LaunchOptions ());`

I tried with below syntax, and error disappears!
`var browser = await Puppeteer.LaunchAsync(new LaunchOptions {});`

**Comment by tarakbpatel** [comment]
OMG...thank you EdgeHeman for giving me this clue. I had the following code:
PuppeteerSharp.LaunchOptions options = new PuppeteerSharp.LaunchOptions { Headless = true };
Which ran fine on my local windows machine and gave me no compilation error. However this was not running on the remote machine that I had to deploy and run my code on until I changed this line to the following:
PuppeteerSharp.LaunchOptions options = new PuppeteerSharp.LaunchOptions() { Headless = true };

Note the parenthesis is needed for some reason. Again there is no compilation error for some reason in the first line and it ran fine on my local machine. I was stuck on this for two weeks...Thank you for the clue.
